### ISSUE TITLE
Add social media icon styling to menu widgets

### ISSUE THREAD

**Issue by melchoyce** [WordPress/twentytwentyone#117]
**Is your feature request related to a problem? Please describe.**
I often want to add social menus to my site. This can be achieved using the beta widget-block screen, but it would also be nice to support social icons inside menu widgets.

**Describe the solution you'd like**
If you link to a social media account within your menu widget, the link is transformed into a social media icon.

**Describe alternatives you've considered**
Only supporting the social icon block, or not including support for social menus — but neither is ideal, IMO.

**Additional context**
Maybe we could reuse the SVGs and styles packaged with the Gutenberg social icon block?

This is a low-priority task.

**Comment by carolinan** [comment]
Is it a navigation menu widget, or an actual navigation menu, located in the footer, that you assign a navigation menu to? 
If the latter, does it replace one of the grids in the footer widget area?

I have no experience of changing the **_navigation menu widget_**. I do not know how it works.

But for a navigation menu that allows both social and regular links, we can just copy back the menu that was already in Seedlet or one of the default themes, and remove the placeholder "link / chain" icon so that regular text is shown if there is no match to a social media site.

**Comment by melchoyce** [comment]
I was thinking it would be a navigation menu widget (and eventually a navigation block), so there's more versatility around placement.

**Comment by beafialho** [comment]
I think this would be a valuable visual addition.

**Comment by carolinan** [comment]
I do not understand the difference between adding a paragraph with two links and a social icon block in a widget area, which is already possible in 5.6?

How is the custom widget offering more flexibility? Both needs the widget area. 
The menu widget has 0 block style options - it is more limited that way. 

Not arguing, trying to understand what needs to be coded. Or if we just need to document how the user can add their social links.

**Comment by melchoyce** [comment]
It sounds like there isn't any guarantee widget blocks will launch with 5.6, unfortunately 😕 That would solve a lot of problems. 

My general thoughts around this:
- We style the navigation menu widget similarly to the primary menu. Horizontal, not vertical.
- If someone includes a link to a popular social media site, we target that link with CSS and change from a label to an icon, similarly to how a lot of themes style social menus.

Let me know if I can clarify more, I hope I'm making sense 😅

**Comment by melchoyce** [comment]
We discussed this in today's [bug scrub](https://wordpress.slack.com/archives/C02RP4VMP/p1601656298186700). We're going to hope widget blocks make it into 5.6, and will chat in our Monday meeting about whether or not it makes sense to add a menu location to the footer. Pending that conversation, we might close this issue.

**Comment by carolinan** [comment]
I am moving this from the milestone because we first need a _final final_ decision on https://github.com/WordPress/twentytwentyone/issues/148
